src/meta/src/manager/streaming_job.rs

```diff
@@ -285,6 +285,45 @@ impl StreamingJob {
         }
     }
 
+    pub fn dependent_secret_refs(&self) -> Vec<u32> {
+        match self {
+            StreamingJob::Sink(sink, _) => {
+                let mut dependents = vec![];
+                for secret_ref in sink.secret_refs.values() {
```

Comment:
This's so weird to extract secret ids from two different places. Are you gonna deprecate one of them and this part is just for backward compatibility? If so, since the secret is not published you can deprecate directly in the proto file.

Reply:
That's probably the most straightforward way to have one secret_refs for connector properties and  one for encode options considering they may have the same key name but ref different secrets.